File: docsapp/com.ibm.concord.document.common/src/com/ibm/concord/document/common/pres/ShapeImg.java
package com.ibm.concord.document.common.pres;

import java.util.Iterator;
import java.util.Map;

import com.ibm.json.java.JSONArray;
import com.ibm.json.java.JSONObject;

public class ShapeImg extends Model
{
  private String divId;

  public ShapeImg(JSONObject json)
  {
    if (!json.isEmpty())
    {
      this.id = json.get("id").toString();
      if (json.containsKey("attrs"))
        this.attrs = (JSONObject) json.get("attrs");
      this.divId = json.get("divId").toString();
    }
  }

  public String getHTML()
  {
    return null;
  }

  public String getDivId()
  {
    return this.divId;
  }

  public String getHTML(JSONArray others)
  {
    String html = "<div";
    html += " id=\"" + this.divId + "\"";
    // TODO: maybe boxId need be added
    html += " class=\"g_draw_frame bc\" style=\"position:absolute;left:0%;top:0%;width:100%;height:100%;\" presentation_class=\"graphic\" text_anchor-type=\"paragraph\" draw_layer=\"layout\" tabindex=\"0\"";
    html += ">";

    String tag = "img";
    html += this.gartherAttrs(tag);
    html += "</" + tag + ">";

    if (others != null)
    {
      for (int i = 0; i < others.size(); i++)
      {
        JSONObject other = (JSONObject) others.get(i);
        String tagName = (String) other.get("tagName");
        String otherHtml = "<" + tagName;
        JSONObject attrs = (JSONObject) other.get("attrs");
        if (attrs != null)
        {
          Iterator it = attrs.entrySet().iterator();
          while (it.hasNext())
          {
            Map.Entry pairs = (Map.Entry) it.next();
            String key = pairs.getKey().toString();
            String value = pairs.getValue().toString();
            String span = " " + key + "=\"" + value + "\"";
            otherHtml += span;
          }
        }
        otherHtml += ">";
        otherHtml += (String) other.get("content");
        otherHtml += "</" + tagName + ">";
        html += otherHtml;
      }
    }

    html += "</div>";
    return html;
  }

}

File: docs-web.resources/com.ibm.docs.web.resources/WebContent/js/presentation/js/model/ShapeImg.js
/* ***************************************************************** */
/*                                                                   */
/* IBM Confidential                                                  */
/*                                                                   */
/* IBM Docs Source Materials                                         */
/*                                                                   */
/* (c) Copyright IBM Corporation 2012. All Rights Reserved.          */
/*                                                                   */
/* U.S. Government Users Restricted Rights: Use, duplication or      */
/* disclosure restricted by GSA ADP Schedule Contract with IBM Corp. */
/*                                                                   */
/* ***************************************************************** */

dojo.provide("pres.model.ShapeImg");

dojo.require("pres.model.Attrable");
dojo.require("pres.model.Htmlable");
dojo.require("pres.utils.helper");

dojo.declare("pres.model.ShapeImg", [pres.model.Attrable, pres.model.Htmlable], {
	constructor: function(json, parent)
	{
		if (parent)
			this.parent = parent;
		this.attrs = {};
		if (json)
		{
			this.id = json.id; // img node id
			this.divId = json.divId; // group frame id
			this.attrs = json.attrs;
		}
	},

	isValid: function()
	{
		return this.id && this.divId;
	},

	clone: function()
	{
		var obj = new pres.model.ShapeImg();
		obj.id = this.id;
		obj.divId = this.divId;
		obj.attrs = dojo.clone(this.attrs);
		return obj;
	},

	toJson: function()
	{
		var json = {
			id: this.id,
			divId: this.divId,
			attrs: dojo.clone(this.attrs)
		};
		return json;
	},

	getHTML: function(others)
	{
		var html = '<div';
		html += ' id="' + this.divId + '"';
		// TODO: maybe boxId need be added
		html += ' class="g_draw_frame bc" style="position:absolute;left:0%;top:0%;width:100%;height:100%;" presentation_class="graphic" text_anchor-type="paragraph" draw_layer="layout" tabindex="0"';
		html += '>';
		
		var tag = 'img';
		html += this._gartherAttrs(null, null, tag);
		html += '</' + tag + '>';
		
		if(others)
		{
			dojo.forEach(others, function(ot){
				
				var tagName = ot.tagName;
				
				var str = tagName ? ("<" + tagName) : "<div";
				var map = ot.attrs;
				for ( var x in map)
				{
					var span = " " + x + "=\"" + map[x] + "\"";
					str += span;
				}
				str += ">";
				str += ot.content || "";
				
				str += tagName ? ("</" + tagName) : "</div";
				str += ">";
				
				html+=str;
			});
		}
		
		html+='</div>';

		return html;
	},
	
	parseDom: function(elemDom)
	{
		var me = this;
		dojo.forEach(elemDom.attributes, function(attr)
		{
			var name = attr.name;
			if (name == "id")
				me.id = attr.value;
			else
			{
				me.attrs[name] = attr.value;
			}
		});
	}
});